### PR TITLE
Add generic font interface + sample (#192)

### PR DESCRIPTION
**Pull request by marekmaskarinec**

### CODE REVIEW

**Comment by vtereshkov** [comment]
@marekmaskarinec Is `canvas::drawText()` removed?

**Comment by marekmaskarinec** [comment]
No, the old api stays for those who want to use it. But now there is also the `canvas::pixelFont` built over it.

File: umka/canvas.um [hunk 1]
```diff
@@ -47,6 +48,25 @@ fn textSize*(text: str, scale: th::fu): th::Vf2 {
 	return {maxw * scale * 6 - scale, h * 6 * scale}
 }
 
+type __PixelFont = struct { }
```
Comment (skejeton) [inline umka/canvas.um:51]: Do we need the double underscore?

File: umka/font.um [hunk 2]
```diff
@@ -14,43 +14,51 @@ type Filter* = enum {
 }
 //~~
 
-//~~opaque Font
-type Font* = struct { _: ^struct{} }
-//~~
+// TODO: should this be exported and documented?
+type TtfFont* = struct { _: ^struct{} }
```
Comment (skejeton) [inline umka/font.um:18]: I think it's OK to export it. This is just a minor nitpick - I think `TtfFont` doesn't look very nice, maybe we can call it `Truetype`.

File: umka/font.um [hunk 3]
```diff
@@ -14,43 +14,51 @@ type Filter* = enum {
...
 		return {}
 	}
 
-	return umth_font_measure(f^, text)
+	return umth_ttf_font_measure(f^, text)
+}
+
+//~~interface Font
+// A generic interface for fonts. Fonts can be loaded from file using
+// `font.load`, or a built in canvas can be used from `canvas.pixelFont`.
```
Comment (skejeton) [inline umka/font.um:46]: Should be `font::load` and `canvas::pixelFont`

**Review by skejeton — CHANGES_REQUESTED** [review]
Looks good! Only minor nitpicks like in documentation and naming. I'm going to test it on SaveScum.

File: umka/canvas.um [hunk 4]
```diff
@@ -4,6 +4,7 @@
 //~~
 
 import (
+	"font.um"
```
Comment (skejeton) [inline umka/canvas.um:7]: This import isn't used here.

**Comment by skejeton** [comment]
Works with no issues.

**Comment by marekmaskarinec** [comment]
@skejeton I have incorporated your suggestions. Are we ok to merge now?

**Pull request merged**